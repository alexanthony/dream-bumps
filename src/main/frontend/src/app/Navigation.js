import React, { PropTypes } from 'react';
import { IndexLink, Link } from 'react-router';
import { connect } from 'react-redux';
import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar';
import FlatButton from 'material-ui/FlatButton';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';
import Divider from 'material-ui/Divider';

import {isAdmin} from '../rootReducer';

export const Navigation = ({ authenticated, isAdmin }) => (
      <Toolbar>
        <ToolbarGroup>
          <FlatButton
            containerElement={<IndexLink to="/" onlyActiveOnIndex />} label="Dream Bumps" style={{color: '#fff'}}
          />
        </ToolbarGroup>
        <ToolbarGroup lastChild>
            { authenticated ?
              <FlatButton
                containerElement={<Link to="/logout" />} label="Log out" style={{color: '#fff'}}
              />
               :
              <FlatButton
                containerElement={<Link to="/login" />} label="Log in" style={{color: '#fff'}}
              />
            }
            { isAdmin &&
              <FlatButton containerElement={<Link to="/docs" />} label="API" style={{color: '#fff'}} />
            }
            { isAdmin &&
              <DropDownMenu value={1} style={{textTransform: 'uppercase', top: '-4px'}} labelStyle={{color: '#fff'}}>
                <MenuItem value={1} label="Manage" primaryText="Users" containerElement={<Link to="/admin/users" />} />
                <MenuItem value={2} label="Manage" primaryText="Crews" containerElement={<Link to="/admin/crews" />}  />
                <MenuItem value={3} label="Manage" primaryText="Market Status" containerElement={<Link to="/admin/market" />}  />
                <Divider />
                <MenuItem value={4} label="Manage" primaryText="Bumps - Men" containerElement={<Link to="/admin/bumps/male" />}  />
                <MenuItem value={5} label="Manage" primaryText="Bumps - Women" containerElement={<Link to="/admin/bumps/female" />}  />
              </DropDownMenu>
            }

            <FlatButton containerElement={<Link to="/bunglines" />} label="Bunglines" style={{color: '#fff'}} />
            <FlatButton containerElement={<Link to="/rankings" />} label="Rankings" style={{color: '#fff'}} />

        </ToolbarGroup>
      </Toolbar>
    );

Navigation.propTypes = {
  authenticated: PropTypes.bool,
  isAdmin: PropTypes.bool
};

export const mapStateToProps = state => ({
  authenticated: state.auth.authenticated,
  isAdmin: isAdmin(state)()
});

export default connect(mapStateToProps)(Navigation);
