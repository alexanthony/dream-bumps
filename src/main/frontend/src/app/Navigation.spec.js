import React from 'react';
import { shallow } from 'enzyme';
import { Navigation, mapStateToProps } from './Navigation';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <Navigation
      authenticated
      isAdmin
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

test('Renders properly', () => {
  const component = shallow(
    <Navigation
      authenticated={false}
      isAdmin={false}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

test('Renders properly', () => {
  const component = shallow(
    <Navigation
      authenticated
      isAdmin={false}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});

test('mapStateToProps', () => {
  const reduxState = {
    auth: {
      authenticated: true,
      currentUser: {
        authorities: ['ROLE_USER']
      }
    }
  };

  const componentStateProps = mapStateToProps(reduxState);

  expect(componentStateProps).toEqual({authenticated: true, isAdmin: false});
});
