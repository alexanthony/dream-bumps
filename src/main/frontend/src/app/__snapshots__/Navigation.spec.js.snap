exports[`test Renders properly 1`] = `
<Toolbar
  noGutter={false}>
  <ToolbarGroup
    firstChild={false}
    lastChild={false}>
    <FlatButton
      containerElement={
        <IndexLink
          onlyActiveOnIndex={true}
          to="/" />
      }
      disabled={false}
      label="Dream Bumps"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
  <ToolbarGroup
    firstChild={false}
    lastChild={true}>
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/logout" />
      }
      disabled={false}
      label="Log out"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/docs" />
      }
      disabled={false}
      label="API"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <DropDownMenu
      animated={true}
      autoWidth={true}
      disabled={false}
      iconButton={<NavigationArrowDropDown />}
      labelStyle={
        Object {
          "color": "#fff",
        }
      }
      maxHeight={500}
      openImmediately={false}
      style={
        Object {
          "textTransform": "uppercase",
          "top": "-4px",
        }
      }
      value={1}>
      <MenuItem
        checked={false}
        containerElement={
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/users" />
        }
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        label="Manage"
        primaryText="Users"
        value={1} />
      <MenuItem
        checked={false}
        containerElement={
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/crews" />
        }
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        label="Manage"
        primaryText="Crews"
        value={2} />
      <MenuItem
        checked={false}
        containerElement={
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/market" />
        }
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        label="Manage"
        primaryText="Market Status"
        value={3} />
      <Divider
        inset={false} />
      <MenuItem
        checked={false}
        containerElement={
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/bumps/male" />
        }
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        label="Manage"
        primaryText="Bumps - Men"
        value={4} />
      <MenuItem
        checked={false}
        containerElement={
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/admin/bumps/female" />
        }
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        label="Manage"
        primaryText="Bumps - Women"
        value={5} />
    </DropDownMenu>
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/bunglines" />
      }
      disabled={false}
      label="Bunglines"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/rankings" />
      }
      disabled={false}
      label="Rankings"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
</Toolbar>
`;

exports[`test Renders properly 2`] = `
<Toolbar
  noGutter={false}>
  <ToolbarGroup
    firstChild={false}
    lastChild={false}>
    <FlatButton
      containerElement={
        <IndexLink
          onlyActiveOnIndex={true}
          to="/" />
      }
      disabled={false}
      label="Dream Bumps"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
  <ToolbarGroup
    firstChild={false}
    lastChild={true}>
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/login" />
      }
      disabled={false}
      label="Log in"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/bunglines" />
      }
      disabled={false}
      label="Bunglines"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/rankings" />
      }
      disabled={false}
      label="Rankings"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
</Toolbar>
`;

exports[`test Renders properly 3`] = `
<Toolbar
  noGutter={false}>
  <ToolbarGroup
    firstChild={false}
    lastChild={false}>
    <FlatButton
      containerElement={
        <IndexLink
          onlyActiveOnIndex={true}
          to="/" />
      }
      disabled={false}
      label="Dream Bumps"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
  <ToolbarGroup
    firstChild={false}
    lastChild={true}>
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/logout" />
      }
      disabled={false}
      label="Log out"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/bunglines" />
      }
      disabled={false}
      label="Bunglines"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/rankings" />
      }
      disabled={false}
      label="Rankings"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "color": "#fff",
        }
      } />
  </ToolbarGroup>
</Toolbar>
`;
